src/main/java/com/google/sps/servlets/SignUpServlet.java

```diff
@@ -0,0 +1,90 @@
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.Key;
+import com.google.cloud.datastore.FullEntity;
+import com.google.cloud.datastore.KeyFactory;
+import com.google.cloud.datastore.Query;
+import com.google.cloud.datastore.QueryResults;
+import com.google.cloud.datastore.StructuredQuery.PropertyFilter;
+import com.google.gson.Gson;
+import com.google.sps.data.User;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.jsoup.Jsoup;
+import org.jsoup.safety.Whitelist;
+		
+@WebServlet("/sign-up")
+public class SignUpServlet extends HttpServlet {
+
+	@Override
+	public void doPost(HttpServletRequest request, HttpServletResponse response) 
+			throws IOException {		    
+		// Get the username, password and email from user
+		 String username = Jsoup.clean(request.getParameter("username"), Whitelist.none());
+		 String password = Jsoup.clean(request.getParameter("password"), Whitelist.none());
+		 String email = Jsoup.clean(request.getParameter("email"), Whitelist.none());
+
+		 Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
+
+		 // Check if username already exists within Datastore
+		 Query<Entity> usernameQuery = Query.newEntityQueryBuilder()
+			 .setKind("User")
+			 .setFilter(PropertyFilter.eq("username", username))
+			 .build();
+		 QueryResults<Entity> usernames = datastore.run(usernameQuery);
+		 boolean usernameExists = false;
+
+		 while(usernames.hasNext() && !usernameExists){
+			 usernameExists = true;
+		 }
```

Comment:
maybe we can move this to a private function, something like checkIfUserExists, to avoid doing all the heavy lifting in one function. same for below to check if the email exists

Reply:
Made changes in 3e571b0